shell/common/shell.cc

```diff
@@ -61,16 +61,18 @@ std::unique_ptr<Shell> Shell::CreateShellOnPlatformThread(
   std::promise<fml::WeakPtr<SnapshotDelegate>> snapshot_delegate_promise;
```

Comment:
If we're going to land this, `Shell::CreateShellOnPlatformThread` should be properly documented by stating that the `Shell` returned may not be ready yet due to asynchronous setup but all functions can be called as normal because we'll defer them in a queue until the shell is set up.

Reply:
Sure thing, only if we've reach agreement on this asynchronous setup idea. And there are many failed test cases to be updated due to asynchronous setup.